oak-segment-azure/src/main/java/org/apache/jackrabbit/oak/segment/azure/AzureStorageCredentialManager.java

```diff
@@ -0,0 +1,146 @@
+    private static final String AZURE_DEFAULT_SCOPE = "https://storage.azure.com/.default";
+    private static final long TOKEN_REFRESHER_INITIAL_DELAY = 45L;
+    private static final long TOKEN_REFRESHER_DELAY = 1L;
+    private ClientSecretCredential clientSecretCredential;
+    private AccessToken accessToken;
+    private StorageCredentialsToken storageCredentialsToken;
+    private final ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
+
+    public StorageCredentials getStorageCredentialsFromEnvironment(@NotNull String accountName, @NotNull Environment environment) {
+        final String clientId = environment.getVariable(AZURE_CLIENT_ID);
+        final String clientSecret = environment.getVariable(AZURE_CLIENT_SECRET);
+        final String tenantId = environment.getVariable(AZURE_TENANT_ID);
+
+        if (StringUtils.isNoneBlank(clientId, clientSecret, tenantId)) {
+            try {
+                return getStorageCredentialAccessTokenFromServicePrincipals(accountName, clientId, clientSecret, tenantId);
+            } catch (IllegalArgumentException | StringIndexOutOfBoundsException e) {
+                throw new IllegalArgumentException(
+                        "Could not connect to the Azure Storage. Please verify if AZURE_CLIENT_ID, AZURE_CLIENT_SECRET and AZURE_TENANT_ID environment variables are correctly set!");
+            }
+        }
+
+        log.warn("AZURE_CLIENT_ID, AZURE_CLIENT_SECRET and AZURE_TENANT_ID environment variables empty or missing. Switching to authentication with AZURE_SECRET_KEY.");
+
+        String key = environment.getVariable(AZURE_SECRET_KEY);
+        try {
+            return new StorageCredentialsAccountAndKey(accountName, key);
+        } catch (IllegalArgumentException | StringIndexOutOfBoundsException e) {
+            throw new IllegalArgumentException(
+                    "Could not connect to the Azure Storage. Please verify if AZURE_SECRET_KEY environment variable is correctly set!");
+        }
+    }
+
+    public StorageCredentials getStorageCredentialAccessTokenFromServicePrincipals(String accountName, String clientId, String clientSecret, String tenantId) {
+        boolean isAccessTokenGenerated = false;
+        if (accessToken == null) {
+            clientSecretCredential = new ClientSecretCredentialBuilder()
+                    .clientId(clientId)
+                    .clientSecret(clientSecret)
+                    .tenantId(tenantId)
+                    .build();
+            accessToken = clientSecretCredential.getTokenSync(new TokenRequestContext().addScopes(AZURE_DEFAULT_SCOPE));
+            if (accessToken == null || StringUtils.isBlank(accessToken.getToken())) {
+                log.error("Access token is null or empty");
+                throw new IllegalArgumentException("Could not connect to azure storage, access token is null or empty");
```

Comment:
I suggest removing the log.error from here, because the exception contains the same information.